agent-bootstrap/src/main/java/io/opentelemetry/auto/bootstrap/instrumentation/decorator/ClientTracer.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.auto.bootstrap.instrumentation.decorator;
+
+import static io.opentelemetry.OpenTelemetry.getPropagators;
+import static io.opentelemetry.trace.TracingContextUtils.getSpan;
+
+import io.grpc.Context;
+import io.opentelemetry.context.propagation.HttpTextFormat;
+import io.opentelemetry.trace.Span;
+import io.opentelemetry.trace.SpanContext;
+
+public abstract class ClientTracer extends BaseTracer {
+
+  public static <C> SpanContext extract(final C carrier, final HttpTextFormat.Getter<C> getter) {
```

Comment:
extract is related to `SERVER` and `CONSUMER` spans, while inject is related to `CLIENT` and `PRODUCER` spans, so not sure having `extract` method in `ClientTracer` makes sense